### PR TITLE
Programs. And test util to create currency.

### PR DESCRIPTION
**Pull request by tjukes**

### CODE REVIEW

File: src/main/java/com/lightrail/LightrailClient.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
     public final Contacts contacts = new Contacts(this);
     public final Transactions transactions = new Transactions(this);
     public final Values values = new Values(this);
+    public final Programs programs = new Programs(this);
```
Comment (pushplay) [inline src/main/java/com/lightrail/LightrailClient.java]: Alphabetical ordering.

File: src/main/java/com/lightrail/Programs.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return lr.networkProvider.getPaginatedList("/programs", Program.class);
+    }
+
+    public PaginatedList<Program> listPrograms(ListProgramsParams params) throws IOException, LightrailRestException {
+        NullArgumentException.check(params, "params");
+
+        return lr.networkProvider.getPaginatedList(String.format("/programs%s", toQueryString(params)), Program.class);
+    }
+
+    public PaginatedList<Program> listPrograms(Map<String, String> params) throws IOException, LightrailRestException {
```
Comment (pushplay) [inline src/main/java/com/lightrail/Programs.java]: Awesome.  I'm glad you caught that I was doing this.  It's sort of a cop-out to deal with the wide variety of param + operator combinations that list endpoints support.

File: src/main/java/com/lightrail/model/Program.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public String id;
+    public String name;
+    public String currency;
+    public Boolean discount;
+    public Boolean pretax;
+    public Boolean active;
+    public Double discountSellerLiability;
+    public Integer minInitialBalance;
+    public Integer maxInitialBalance;
+    public Integer[] fixedInitialBalances;
```
Comment (pushplay) [inline src/main/java/com/lightrail/model/Program.java]: Use `List<Integer>` instead.  Arrays are oldschool in Java and represent a fixed length of memory.  They're mostly used to implement Collections like List.

File: src/main/java/com/lightrail/model/Program.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    public BalanceRule balanceRule;
+    public Date startDate;
+    public Date endDate;
+    public Map<String, Object> metadata;
+    public Date createdDate;
+    public Date updatedDate;
+    public String createdBy;
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (pushplay) [inline src/main/java/com/lightrail/model/Program.java]: Excellent.

File: src/test/java/com/lightrail/ProgramsTest.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        assertEquals(params.pretax, programCreated.pretax);
+        assertEquals(params.redemptionRule, programCreated.redemptionRule);
+        assertEquals(params.minInitialBalance, programCreated.minInitialBalance);
+        assertEquals(params.maxInitialBalance, programCreated.maxInitialBalance);
+        assertEquals(params.fixedInitialUsesRemaining, programCreated.fixedInitialUsesRemaining);
+        assertEquals(params.metadata, programCreated.metadata);
+        assertNotNull(programCreated.createdDate);
+        assertNotNull(programCreated.updatedDate);
+
+        System.out.println(programCreated.active);
```
Comment (pushplay) [inline src/test/java/com/lightrail/ProgramsTest.java]: Clean up.

File: src/main/java/com/lightrail/Programs.java (hunk 2 above)
Comment (tjukes) [reply src/main/java/com/lightrail/Programs.java]: I'm adding a test for it too.

**Pull request merged**